src/main/java/com/eprosima/fastdds/idl/templates/TypeObjectTestingTestSource.stg

```diff
@@ -1096,8 +1096,7 @@ if (type_ids.type_identifier1()._d() == TI_PLAIN_ARRAY_SMALL)
     if (type_ids.type_identifier2()._d() == TK_NONE)
     {
         EXPECT_EQ(type_ids.type_identifier1().array_sdefn().header().equiv_kind(), EK_BOTH);
-        EXPECT_TRUE(element_type_ids.type_identifier1() == *type_ids.type_identifier1().array_sdefn().element_identifier() ||
-            element_type_ids.type_identifier2() == *type_ids.type_identifier1().array_sdefn().element_identifier());
+        EXPECT_TRUE(element_type_ids.type_identifier1() == *type_ids.type_identifier1().array_sdefn().element_identifier());
```

Comment:
Nitpick: for the sake of completeness I suggest adding checking that the second element TypeIdentifier is TK_NONE. This suggestion if accepted, also applies to the other similar checks.

Reply:
Added in [bc87e17](https://github.com/eProsima/Fast-DDS-Gen/pull/307/commits/bc87e1748161f8e3fa66699a2527ab8b90cc8934)